lightningd/jsonrpc.c

```diff
@@ -304,9 +304,17 @@ static void json_add_help_command(struct command *cmd,
 				  struct json_command *json_command)
 {
 	char *usage;
+	struct json_command *tmp;
 	cmd->mode = CMD_USAGE;
+
+	// Temporarily swap the json_cmd pointer, so the dispatch
+	// method can set cmd->usage properly based on the actual
+	// json_command usage
+	tmp = cmd->json_cmd;
```

Comment:
It's probably better to create a clone of `cmd` instead of modifying patching it in place, and then unpatching it.